src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -955,33 +955,104 @@ private IEnumerable<Expression> CompileInvocationArguments(IEnumerable<Expressio
             return arguments == null ? Array.Empty<Expression>() : arguments.Select(CompileExpressionOperand);
         }
 
-        internal Expression ReduceAssignment(ISupportsAssignment left, TokenKind tokenKind, Expression right)
+        internal Expression ReduceAssignment(
+            ISupportsAssignment left,
+            TokenKind tokenKind,
+            Expression right,
+            MemberExpressionAst lhsMemberAccessAst)
         {
-            IAssignableValue av = left.GetAssignableValue();
-            ExpressionType et = ExpressionType.Extension;
+            IAssignableValue assignmentTarget = left.GetAssignableValue();
+            var expressionType = ExpressionType.Extension;
 
             switch (tokenKind)
             {
-                case TokenKind.Equals: return av.SetValue(this, right);
-                case TokenKind.PlusEquals: et = ExpressionType.Add; break;
-                case TokenKind.MinusEquals: et = ExpressionType.Subtract; break;
-                case TokenKind.MultiplyEquals: et = ExpressionType.Multiply; break;
-                case TokenKind.DivideEquals: et = ExpressionType.Divide; break;
-                case TokenKind.RemainderEquals: et = ExpressionType.Modulo; break;
-                case TokenKind.QuestionQuestionEquals: et = ExpressionType.Coalesce; break;
+                case TokenKind.Equals:
+                    return assignmentTarget.SetValue(this, right);
```

Comment:
Since you refactor the code you could move the `Equals` in separate `if` block and convert the switch to a switch expression.